src/lm_buddy/configs/jobs/hf_evaluate.py

```diff
@@ -0,0 +1,83 @@
+from pydantic import Field, conlist, field_validator, model_validator
+
+from lm_buddy.configs.common import LMBuddyConfig
+from lm_buddy.configs.huggingface import (
+    AutoModelConfig,
+    AutoTokenizerConfig,
+    DatasetConfig,
+    QuantizationConfig,
+)
+from lm_buddy.configs.jobs.common import JobConfig
+from lm_buddy.configs.vllm import VLLMCompletionsConfig
+from lm_buddy.paths import AssetPath
+
+
+class HuggingFaceEvaluationConfig(LMBuddyConfig):
+    """Misc settings provided to an lm-harness evaluation job."""
+
+    metrics: conlist(str, min_length=1)
+    use_pipeline: bool = False
+    enable_tqdm: bool = False
+    max_samples: int | None = None
+    storage_path: str | None = None
+
+
+class HuggingFaceEvalJobConfig(JobConfig):
+    """Configuration to run a HuggingFace evaluation job."""
+
+    dataset: DatasetConfig = Field(
+        description="Dataset of text completions to evaluate using the Prometheus judge model."
+    )
+    evaluation: HuggingFaceEvaluationConfig
+    model: AutoModelConfig | VLLMCompletionsConfig
+    quantization: QuantizationConfig | None = None
+    tokenizer: AutoTokenizerConfig
+
+    @model_validator(mode="before")
+    def ensure_tokenizer_config(cls, values):
+        """Set the tokenizer to the model path when not explicitly provided."""
+        if values.get("tokenizer") is None:
```

Comment:
We might have previous code for this that we can use here instead? https://github.com/mozilla-ai/lm-buddy/blob/72d617963f0f83c568d32615d59d3f30403d6b5f/src/lm_buddy/jobs/asset_loader.py#L135

Reply:
Whoops, not sure I understood your comment: we do call `load_pretrained_tokenizer` in the entrypoint code, `ensure_tokenizer_config` just makes sure there is always an `AutoTokenizerConfig` to pass even when there is none provided (in that case it will just use the model path as the tokenizer path).
WDYT about the choice of constraining the config class to `AutoTokenizerConfig`? I appreciate the strict typing even if it requires some extra validation as the above, but what I am not 100% happy with is that, depending on the application, we might have some cases where we do not need a tokenizer at all (e.g. when using an openai-compatible API) and we still have to go through generating one.

Reply:
Yep, that makes sense - my comment was more about the fact that we do this check several times in the platform IIRC and might make sense to have a utility function, but can be addressed in a follow-up

Reply:
Great, thanks! 🙏